File: #3/Categoria 4/swap.js
const swap = (xs, ini, end) => {
  if((xs.length -1) < end) return xs
  const aux = xs[ini]
  xs[ini] = xs[end]
  xs[end] = aux

  return xs 
}

console.log(swap([], 0, 5))
console.log(swap([1], 0, 3))
console.log(swap([1,3,4], 1, 2))
console.log(swap([1,2,3,4,5,6], 2, 5))
console.log(swap([5,6,7,8,9], 0, 3))

File: #3/Categoria 4/swap.hs
swap xs p q 
  | p < size && q < size = left ++ (xs !! q):miolo ++ (xs !! p):right
  | otherwise = xs 
  where 
    size = length xs
    left = take p xs
    right = drop (q + 1) xs
    miolo = take (q - p - 1) (drop (p + 1) xs)

main = do
    a <- readLn :: IO [Int]
    b <- readLn :: IO Int
    c <- readLn :: IO Int
    print $ swap a b c